Observer/BackfillTransactions.php

```diff
@@ -15,309 +15,321 @@
-        Logger $logger,
-        OrderRepositoryInterface $orderRepository,
-        FilterBuilder $filterBuilder,
-        SearchCriteriaBuilder $searchCriteriaBuilder,
-        TaxjarConfig $taxjarConfig,
+        \Magento\Framework\App\RequestInterface $request,
+        \Taxjar\SalesTax\Model\Logger $logger,
+        \Magento\Sales\Api\OrderRepositoryInterface $orderRepository,
+        \Magento\Store\Model\StoreManagerInterface $storeManager,
+        \Magento\Framework\Api\SearchCriteriaBuilder $searchCriteriaBuilder,
         \Magento\Framework\Bulk\BulkManagementInterface $bulkManagement,
         \Magento\AsynchronousOperations\Api\Data\OperationInterfaceFactory $operationFactory,
         \Magento\Framework\DataObject\IdentityGeneratorInterface $identityService,
         \Magento\Framework\Serialize\SerializerInterface $serializer,
-        \Magento\Authorization\Model\UserContextInterface $userContext
+        \Magento\Authorization\Model\UserContextInterface $userContext,
+        \Taxjar\SalesTax\Model\Configuration $taxjarConfig
     ) {
         $this->request = $request;
-        $this->storeManager = $storeManager;
-        $this->logger = $logger->setFilename(TaxjarConfig::TAXJAR_TRANSACTIONS_LOG)->force();
+        $this->logger = $logger->setFilename(\Taxjar\SalesTax\Model\Configuration::TAXJAR_TRANSACTIONS_LOG)->force();
         $this->orderRepository = $orderRepository;
-        $this->filterBuilder = $filterBuilder;
+        $this->storeManager = $storeManager;
         $this->searchCriteriaBuilder = $searchCriteriaBuilder;
-        $this->taxjarConfig = $taxjarConfig;
         $this->bulkManagement = $bulkManagement;
         $this->operationFactory = $operationFactory;
-        $this->identityService = $identityService;
         $this->serializer = $serializer;
         $this->userContext = $userContext;
+        $this->taxjarConfig = $taxjarConfig;
+        $this->uuid = $identityService->generateId();
     }
 
     /**
-     * @param Observer $observer
-     * @throws LocalizedException
+     * @throws \Magento\Framework\Exception\LocalizedException
      */
-    public function execute(Observer $observer): void
+    public function execute(\Magento\Framework\Event\Observer $observer)
     {
-        $this->apiKey = $this->taxjarConfig->getApiKey();
+        $this->observer = $observer;
 
-        if (!$this->apiKey) {
-            throw new LocalizedException(
-                __('Could not sync transactions with TaxJar. Please make sure you have an API key.')
-            );
+        try {
+            if (!$this->taxjarConfig->getApiKey()) {
+                throw new \Magento\Framework\Exception\LocalizedException(
+                    __('Could not sync transactions with TaxJar. Please make sure you have an API key.')
+                );
+            }
+
+            $criteria = $this->getSearchCriteria();
+            $orders = $this->getOrders($criteria, $this->getInput('force'));
```

Comment:
When running a backfill from the UI without force enabled I am getting the following error:

Argument 2 passed to Taxjar\SalesTax\Observer\BackfillTransactions::getOrders() must be of the type bool or null, string given, called in /var/www/html/app/code/Taxjar/SalesTax/Observer/BackfillTransactions.php on line 135

It looks like $this->getInput('force') is returning the string '0'

Reply:
Ahh crap.. nice. Think I might've added that bool type while writing tests.

Reply:
Added casts to bool and updated unit tests to account for this scenario